tests_e2e/tests/lib/agent_test.py

```diff
@@ -94,3 +101,16 @@ def _run_remote_test(self, command: str, use_sudo: bool = False, attempts: int =
     @staticmethod
     def _indent(text: str, indent: str = " " * 8):
         return "\n".join(f"{indent}{line}" for line in text.splitlines())
+
+
+class AgentVmTest(AgentTest, ABC):
```

Comment:
what is ABC here?

Reply:
Standard python class, "Abstract Base Class". You combine this with the abstractmethod decorator and if you try to instantiate one of these classes without providing an implementation for all abstract methods, Python raises an exception.

It is actually not needed here, AgentTest is already an ABC. Will remove it.